File: oss-center/oss-center-bootstrap/src/main/java/com/jmsoftware/maf/osscenter/OssCenterApplication.java
package com.jmsoftware.maf.osscenter;

import com.jmsoftware.maf.springcloudstarter.helper.SpringBootStartupHelper;
import lombok.extern.slf4j.Slf4j;
import lombok.val;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
import org.springframework.cloud.openfeign.EnableFeignClients;
import org.springframework.util.StopWatch;

/**
 * <h1>OssCenterApplication</h1>
 * <p>
 * Change description here.
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com, date: 6/20/21 1:32 PM
 **/
@Slf4j
@EnableFeignClients
@EnableDiscoveryClient
@SpringBootApplication
@SuppressWarnings("scwjava_Createprivateconstructorforutilityclassallfieldsmethodsarestatic")
public class OssCenterApplication {
    public static void main(String[] args) {
        val stopWatch = new StopWatch();
        stopWatch.start();
        val configurableApplicationContext = SpringApplication.run(OssCenterApplication.class, args);
        val springBootStartupHelper = configurableApplicationContext.getBean(SpringBootStartupHelper.class);
        springBootStartupHelper.stop(stopWatch);
    }
}

File: oss-center/oss-center-bootstrap/src/main/kotlin/com/jmsoftware/maf/osscenter/OssCenterApplication.kt
package com.jmsoftware.maf.osscenter

import com.jmsoftware.maf.springcloudstarter.helper.SpringBootStartupHelper
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.cloud.client.discovery.EnableDiscoveryClient
import org.springframework.cloud.openfeign.EnableFeignClients
import org.springframework.util.StopWatch

/**
 * # OssCenterApplication
 *
 * Change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/17/22 8:47 AM
 */
@EnableFeignClients
@EnableDiscoveryClient
@SpringBootApplication
class OssCenterApplication

fun main(args: Array<String>) {
    StopWatch().let {
        it.start()
        runApplication<OssCenterApplication>(*args)
            .getBean(SpringBootStartupHelper::class.java)
            .stop(it)
    }
}
